tests/unittests/Foundation/NSProxyTests.mm

```diff
@@ -0,0 +1,161 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import <Foundation/Foundation.h>
+#import <TestFramework.h>
+
+// Ignore warnings about functions in this test class not having method definitions - they work through forwarding to forwardArray
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wincomplete-implementation"
+
+// A basic NSProxy for testing that forwards to a NSMutableArray without retaining it
+@interface _NSTestMutableArrayProxy : NSProxy
+- (instancetype)initWithMutableArray:(NSMutableArray*)array;
+@end
+
+@interface _NSTestMutableArrayProxy ()
+@property (readonly, assign) NSMutableArray* forwardArray; // Not retained
+
+// NSArray functions that are forwarded (Suppresses 'may not respond to' warnings)
+- (NSUInteger)count;
+- (id)objectAtIndex:(NSUInteger)index;
+- (void)addObject:(id)anObject;
+- (void)insertObject:(id)anObject atIndex:(NSUInteger)index;
+@end
+
+@implementation _NSTestMutableArrayProxy
+
+- (instancetype)initWithMutableArray:(NSMutableArray*)array {
+    self->_forwardArray = array;
```

Comment:
Indeed! It'd be good to have a comment in here so somebody doesn't think it's broken.
Our CC (or that one really didactic talk I gave) states that we should _almost never_ not call init. Since this is one of the cases where we mustn't, it deserves callout!